mlrun/runtimes/serving.py

```diff
@@ -13,306 +13,199 @@
-        return request
-
-    def postprocess(self, request: Dict) -> Dict:
-        return request
-
-    def predict(self, request: Dict) -> Dict:
-        raise NotImplementedError
-
-    def explain(self, request: Dict) -> Dict:
-        raise NotImplementedError
-
-
-def nuclio_serving_init(context, data):
-    model_prefix = "SERVING_MODEL_"
-    params_prefix = "SERVING_PARAMS"
-
-    # Initialize models from environment variables
-    # Using the {model_prefix}_{model_name} = {model_path} syntax
-    model_paths = {
-        k[len(model_prefix) :]: v
-        for k, v in os.environ.items()
-        if k.startswith(model_prefix)
-    }
-    model_class = os.environ.get("MODEL_CLASS", "MLModelServer")
-    fhandler = data[model_class]
-    models = {
-        name: fhandler(name=name, model_dir=path) for name, path in model_paths.items()
-    }
-
-    params = os.environ.get(params_prefix)
-    if params:
-        params = json.loads(params)
-
-    for name, model in models.items():
-        if params:
-            setattr(model, "_params", deepcopy(params))
-        if not model.ready:
-            model.load()
-            model.ready = True
-
-    # Verify that models are loaded
-    assert len(models) > 0, (
-        "No models were loaded!\n Please load a model by using the environment variable "
-        "SERVING_MODEL_{model_name} = model_path"
-    )
-
-    context.logger.info(f"Loaded {list(models.keys())}")
-
-    # Initialize route handlers
-    hostname = socket.gethostname()
-    server_context = _ServerInfo(context, hostname, model_class)
-    predictor = PredictHandler(models).with_context(server_context)
-    explainer = ExplainHandler(models).with_context(server_context)
-    router = {"predict": predictor.post, "explain": explainer.post}
-
-    # Define handle
-    setattr(context, "mlrun_handler", nuclio_serving_handler)
-    setattr(context, "models", models)
-    setattr(context, "router", router)
-
-
-err_string = (
-    "Got path: {} \n Path must be <model-name>/<action> \nactions: {} \nmodels: {}"
-)
-
-
-def nuclio_serving_handler(context, event):
-
-    # check if valid route & model
-    try:
-        if hasattr(event, "trigger") and event.trigger.kind != "http":
-            # non http triggers (i.e. stream) are directed to the first model
-            # todo: take model name and action from json is specified
-            model_name = next(iter(context.models))
-            route = "predict"
-        else:
-            model_name, route = event.path.strip("/").split("/")
-        route = context.router[route]
-    except Exception:
-        return context.Response(
-            body=err_string.format(
-                event.path,
-                "|".join(context.router.keys()),
-                "|".join(context.models.keys()),
-            ),
-            content_type="text/plain",
-            status_code=404,
+import nuclio
+
+from .function import RemoteRuntime, NuclioSpec
+
+serving_subkind = "v2serving"
```

Comment:
its not specific for v2, rather to everything post v1 which uses a separate `serving` runtime and init method, i will change the string to `serving_v2`

Reply:
It's a subkind that represent the v2 of serving no ? what's not specific to v2 ?
if tomorrow we'll have a new version it will be `serving_v3_subkind` right ? so makes sense that it will be `serving_v2_subkind` no ?
am I missing anything ?

Reply:
Also you didn't move it from here, it's much more nice to have all constants in one place

Reply:
@Hedingber `nice` is a matter of taste, if i put in init its a circular dep, and there is no reason to do it since its the subkind for the serving runtime (vs the subkind which predated having a specific runtime)